lib/src/rules/unreachable_from_main.dart

```diff
@@ -163,6 +180,22 @@ class _ReferenceVisitor extends RecursiveAstVisitor {
     super.visitClassDeclaration(node);
   }
 
+  @override
+  void visitConstantPattern(ConstantPattern node) {
+    var expression = node.expression;
+    if (expression is InstanceCreationExpression) {
+      expression.argumentList.accept(this);
+      // Intentionally do not visit `expression.constructorName`, as a reference
+      // in a constant pattern is not good enough to count as "reachable".
+      // Marking a type as reachable only because it was seen in a constant
+      // pattern would be a miscategorization if the type is never instantiated
+      // or subtyped.
```

Comment:
Technically, this will instantiate the class, and the object being matched might even be equal to the instance even if it isn't an instance of the same class. I'm not sure that this should be skipped.

Reply:
Hmm wait, can you explain that more? Can the constant being instantiated be equal to an object that wasn't instantiated the same way? Maybe I just need tests with factory constructors?

Reply:
I might be wrong. Does the constant have to use the implementation of `==` from `Object` (that is, `identical`)? I was thinking that it would be possible for the `==` to return `true` for an object other than the same constant, but maybe that's not true.

But in any case, an instance of the class is actually created so that it can be compared to the value being switched over.

Reply:
Good catch!! A constant pattern which is a const invocation is totally valid code there, and does instantiate an object. I've reverted this bit.